lib/Net/SAML2/IdP.pm

```diff
@@ -185,6 +185,8 @@ sub new_from_xml {
     {
         my $use = $key->getAttribute('use') || 'signing';
 
+        my @certs = ();
```

Comment:
I would make this a hashref with $use for the keys and than an array, that makes the logic easier I think. It means `$self->certs->{signing}` will really easy to accomplish.